src/corehost/common/pal.windows.cpp

```diff
@@ -198,16 +182,36 @@ bool pal::get_default_servicing_directory(string_t* recv)
 bool pal::get_global_dotnet_dirs(std::vector<pal::string_t>* dirs)
 {
     pal::string_t dir;
-    if (!get_file_path_from_env(_X("ProgramFiles"), &dir))
+    if (!get_default_installation_dir(&dir))
     {
         return false;
     }
 
-    append_path(&dir, _X("dotnet"));
     dirs->push_back(dir);
     return true;
 }
 
+bool pal::get_default_installation_dir(pal::string_t* recv)
+{
+    pal::char_t* program_files_dir;
+    if (pal::is_running_in_wow64())
+    {
+        program_files_dir = _X("ProgramFiles(x86)");
```

Comment:
This fixes an existing issue with multi-level lookup not working with x86 in wow mode (it would try to look in the x64's Program Files directory). The same wow check is used for DOTNET_ROOT(x86)